### ISSUE TITLE
[BUG] equals on avx

### ISSUE THREAD

**Issue by DenisYaroshevskiy** [jfalcou/eve#728]
equal on avx looks pretty bad:
https://gcc.godbolt.org/z/3MY48qz4E

I'm curious if we can make use of a float point one: _mm256_cmp_ps
Or the xor trick.

**Comment by jfalcou** [comment]
Like: https://gcc.godbolt.org/z/vq3vx84nE ?

**Comment by DenisYaroshevskiy** [comment]
For logicals this is a good idea!

For `wide<int>` we need something else though: https://gcc.godbolt.org/z/KcnTMczcG
F

**Comment by jfalcou** [comment]
https://gcc.godbolt.org/z/aYn3EK8Gx  ?

**Comment by DenisYaroshevskiy** [comment]
@jfalcou does not look good: https://gcc.godbolt.org/z/jK834qnoY
But maybe you can make it work

```
        vxorps  ymm1, ymm0, ymm1
        vpxor   xmm2, xmm2, xmm2
        vmovdqa xmm0, xmm1
        vextractf128    xmm1, ymm1, 0x1
        vpcmpeqd        xmm0, xmm2, xmm0
        vpcmpeqd        xmm2, xmm2, xmm1
        vinsertf128     ymm0, ymm0, xmm2, 0x1
        vxorps  ymm0, ymm0, YMMWORD PTR .LC0[rip]
```

**Comment by jfalcou** [comment]
https://gcc.godbolt.org/z/jndaWWa1z

```c++
eq(eve::wide<int, eve::fixed<8l>, eve::x86_256_>, eve::wide<int, eve::fixed<8l>, eve::x86_256_>):
        vxorps  ymm0, ymm0, ymm1
        vxorps  xmm1, xmm1, xmm1
        vcmpps  ymm0, ymm0, ymm1, 4
        ret
```

**Comment by jfalcou** [comment]
Fixed in b869914
